Template/Tag/FacebookPixelTag.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Plugins\TagManager\Template\Tag;
+
+use Piwik\Settings\FieldConfig;
+use Piwik\Plugins\TagManager\Template\Tag\BaseTag;
+use Piwik\Validators\NotEmpty;
+
+class FacebookPixelTag extends BaseTag
+{
+
+    Public function getName()
+    {
+        return 'Facebook Pixel';
+    }
+
+    public function getDescription()
+    {
+        return 'The Facebook Pixel is a web analytics and avertising service offered by Facebook.';
+    }
+
+    public function getHelp()
+    {
+        return 'This tag lets you track website pageviews in your Facebook Ads account. To obtain the  Pixel ID please log in to your Facebook Ads account.';
+    }
+
+    public function getCategory()
+    {
+        return self::CATEGORY_ANALYTICS;
+    }
+
+    public function getParameters()
+    {
+        return array(
+            $this->makeSetting('pixelId', '', FieldConfig::TYPE_STRING, function (FieldConfig $field) {
+                $field->title = 'Pixel ID';
+                $field->description = 'Long numerical number for Facebook Pixel Code"';
+                $field->validators[] = new NotEmpty();
+                $field->validate = function ($value) {
```

Comment:
I think you can replace this with a `new NumberRange();` (from `Piwik\Validators\NumberRange`). 
That way the message would also be localized.

Reply:
I'm not sure on the minimum/maximum number of digits so think the changes I've made should do for now.

Reply:
The parameter is optional, so this should be good enough.